src/Authentication/Authentication/Cmdlets/ConnectMgGraph.cs

```diff
@@ -170,9 +182,10 @@ private async Task ProcessRecordAsync()
                             _cancellationTokenSource.CancelAfter(authTimeout);
                             authContext.AuthType = AuthenticationType.Delegated;
                             string[] processedScopes = ProcessScopes(Scopes);
-                            authContext.Scopes = processedScopes.Length == 0 ? new string[] { "User.Read" } : processedScopes;
+                            authContext.Scopes = processedScopes.Length == 0 ? new[] { "User.Read" } : processedScopes;
                             // Default to CurrentUser but allow the customer to change this via `ContextScope` param.
                             authContext.ContextScope = this.IsParameterBound(nameof(ContextScope)) ? ContextScope : ContextScope.CurrentUser;
+                            authContext.UseDeviceAuth = UseDeviceAuthentication;
```

Comment:
Consider using an enum instead. e.g:
``` powershell
# UserParameterSet
authContext.AuthProviderType = UseDeviceAuthentication ? AuthProviderType.DeviceCodeProvider : AuthProviderType.InteractiveAuthenticationProvider;
# AppParameterSet
authContext.AuthProviderType = AuthProviderType.ClientCredentialProvider;
# AccessTokenParameterSet
authContext.AuthProviderType = AuthProviderType.None;
```
This will make it easier to retreive the right clinetApplication in AuthenticateAsync.